galaxyproject/galaxy
# Adding fuzzy search functionality to front end search

Why: hackathon project: When searching for a tool using the front end search, the user should be able to make small typos and still receive search results.

What: On tool search an exact substring match is checked to see if a tool matches. If this fails, a modified form of the Damerau–Levenshtein distance algorithm is applied to to find similar results. 

[Galaxy.webm](https://user-images.githubusercontent.com/62220253/228048996-24b3889a-1bbe-4c08-8d37-8d55926edcef.webm)


## How to test the changes?

  1. Click the tool search bar to enter a search
  2. Search for a tool name with both correct and incorrect spelling to see how it appears

## License
